doc/R.nvim.txt

```diff
@@ -549,6 +550,78 @@ put in your `init.lua`:
    vim.g.rout_follow_colorscheme = true
 <
 
+------------------------------------------------------------------------------
+4.7. Debugging R functions (Unix only)                     *debug-R-functions*
+
+The R-Nvim-Plugin has limited debugging facilities available on Unix systems
+(not available on Windows). To debug a function with R-Nvim you have to do the
+following:
+
+   1. Source the R script where the function is with the `source()` function
+      of the `base` package.
+
+   2. In the R Console, run either `debug(funcname)` or `debugonce(funcname)`
+      where `funcname` is the name of the function (not quoted). R-Nvim will
+      send this command to R Console if you press <LocalLeader>db in Normal
+      mode with the cursor over the function name.
+
+   3. Call the function.
+
+See `:Rh browser` for available commands in debugging mode.
+
+For example, if you have a file named `my_function.R` with the following
+contents:
+>r
+   add.one <- function(x){
+       y <- x + 1
+       print(y)
+       return(invisible(NULL))
+   }
+<
+You could debug the `add.one()` function from another script such as:
+>r
+   source("path/to/my_function.R")
+   debug(add.one)
+   add.one(7)
+<
+If you do not want to start debugging the function from its first line,
+instead of calling `debug(funcname)` before calling the function, you should
+add the line `browser()` to the function at the line where the debugging
+session should start. Example:
+>r
+   add.one <- function(x){
+       y <- x + 1
+       browser()
+       print(y)
+       return(invisible(NULL))
+   }
+<
+If you want a conditional breakpoint, you should call the `browser()` command
+only if a condition is satisfied. Example:
+>r
+   add.one <- function(x){
+       y <- x + 1
+       if(x > 9)
+           browser()
+       print(y)
+       return(invisible(NULL))
+   }
+<
+One limitation is that R-Nvim also does not help you setting breakpoints.
+You have to set them manually as explained above.
+
+Another limitation of R-Nvim is that it might be unable to jump to the line in
+the source code that is being evaluated if either you have sent the function
+directly to the R Console or the function has been sent by the `knitr`
+package, without calling the function `source()`. In either case, R-Nvim will
+try to find the pattern `funcname <- function(` in the buffer being edited. If
+the pattern is found, the cursor will jump to its position.
```

Comment:
Seems like it was already a known implementation? (I didn't write the documentation in this PR, I am merely restoring it from the previous state).

Reply:
I just tried sourcing the function in a file using the `source` function and I get the same behaviour (no highlighting). So perhaps the documentation could be updated to reflect that loading functions into the environment via `devtools` will work but not using `source`?

Reply:
If I understand correctly, debugging will only work when sourcing a package with `devtools`.